plugins/org.eclipse.fordiac.ide.ui/src/org/eclipse/fordiac/ide/ui/contentoutline/MultiPageEditorContentOutlinePage.java

```diff
@@ -0,0 +1,241 @@
+				multiPageSite.activate();
+			}
+			currentPage.page().addSelectionChangedListener(selectionChangedListener);
+			if (currentPage.page() instanceof final IPostSelectionProvider postSelectionProvider) {
+				postSelectionProvider.addPostSelectionChangedListener(postSelectionChangedListener);
+			}
+			if (getSite().getActionBars() instanceof final SubActionBars parentSubActionBars) {
+				parentSubActionBars.activate(); // make sure parent is activated
+			}
+			getSite().getActionBars().updateActionBars();
+		}
+		fireSelectionChanged(new SelectionChangedEvent(this, getSelection()));
+		firePostSelectionChanged(new SelectionChangedEvent(this, getSelection()));
+	}
+
+	protected void pageChanged(final PageChangedEvent event) {
+		if (event.getSelectedPage() instanceof final IEditorPart activeEditor) {
+			showPage(getOrCreatePage(activeEditor));
+		} else {
+			showPage(getOrCreatePage(editorPart));
+		}
+	}
+
+	@Override
+	public ISelection getSelection() {
+		if (currentPage != null) {
+			final ISelection selection = currentPage.page().getSelection();
+			if (selection != null) {
+				return selection;
+			}
+		}
+		return StructuredSelection.EMPTY;
+	}
+
+	@Override
+	public void setSelection(final ISelection selection) {
+		if (currentPage != null) {
+			currentPage.page().setSelection(selection);
+		}
+	}
+
+	@Override
+	public void addSelectionChangedListener(final ISelectionChangedListener listener) {
+		selectionChangedListeners.add(listener);
+	}
+
+	@Override
+	public void removeSelectionChangedListener(final ISelectionChangedListener listener) {
+		selectionChangedListeners.remove(listener);
+	}
+
+	@Override
+	public void addPostSelectionChangedListener(final ISelectionChangedListener listener) {
+		postSelectionChangedListeners.add(listener);
+	}
+
+	@Override
+	public void removePostSelectionChangedListener(final ISelectionChangedListener listener) {
+		postSelectionChangedListeners.remove(listener);
+	}
+
+	protected void selectionChanged(final SelectionChangedEvent event) {
+		fireSelectionChanged(new SelectionChangedEvent(this, event.getSelection()));
+	}
+
+	protected void postSelectionChanged(final SelectionChangedEvent event) {
+		firePostSelectionChanged(new SelectionChangedEvent(this, event.getSelection()));
+	}
+
+	protected void fireSelectionChanged(final SelectionChangedEvent event) {
+		selectionChangedListeners.forEach(listener -> SafeRunner.run(() -> listener.selectionChanged(event)));
+	}
+
+	protected void firePostSelectionChanged(final SelectionChangedEvent event) {
+		postSelectionChangedListeners.forEach(listener -> SafeRunner.run(() -> listener.selectionChanged(event)));
+	}
+
+	@Override
+	public Control getControl() {
+		return book;
+	}
+
+	@Override
+	public void setFocus() {
+		if (book != null && !book.isDisposed()) {
+			book.setFocus();
+		}
+		if (currentPage != null) {
+			currentPage.page().setFocus();
+		}
+	}
+
+	public <T> T getAdapter(final Class<T> key) {
```

Comment:
this method is not used. shell the outlinepage implement adaptable?

Reply:
Ah, yes, thanks for noticing!